### PR TITLE
tracing: add an example of `tracing` in a panic hook

### PR DESCRIPTION
**Pull request by hawkw**
It turns out that when using the global dispatcher, emitting tracing
events in a panic hook will capture the span in which the program
panicked. This is very handy for debugging panics! Thanks a lot to
@natemara for pointing this out to me on twitter --- I hadn't even
thought of it!

Since it isn't necessarily immediately obvious that this works, I
thought it would be nice to add an example.

### CODE REVIEW

**Review by davidbarsky — APPROVED** [review]
woah, this is rad!

File: examples/examples/panic_hook.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    // torn down by unwinding by the time the panic handler is reached.
+    tracing::collect::set_global_default(collector).unwrap();
+
+    // Set a panic hook that records the panic as a `tracing` event at the
+    // `ERROR` verbosity level.
+    //
+    // If we are currently in a span when the panic occurred, the logged event
+    // will include the current span, allowing the context in which the panic
+    // occurred to be recorded.
+    std::panic::set_hook(Box::new(|panic| {
```
Comment (LucioFranco) [inline examples/examples/panic_hook.rs:23]: Can we make this built-in?

```
tracing::set_panic_hook()
```

and could fail if no global one is set?

Comment (davidbarsky) [reply examples/examples/panic_hook.rs:23]: wow, i finally get to drop this link: https://github.com/tokio-rs/tracing/issues/new?assignees=&labels=&template=feature_request.md

Comment (hawkw) [reply examples/examples/panic_hook.rs:23]: yes, might be worth a follow-up change to add something like that.

Comment (davidbarsky) [reply examples/examples/panic_hook.rs:23]: I'd suggest opening an issue to centralize discussion: https://github.com/tokio-rs/tracing/issues/new?assignees=&labels=&template=feature_request.md.

However, I'm personally _not_ inclined to have this in `tracing` for the following reasons:
- Users might want to handle panics in different ways, with different formatting, levels, and so forth. They might not want to involve tracing in the first place!
- Other users might not want `tracing` to change their panic hooks _or_ they might want `tracing` _and_ some other functionality in their panic hooks. If the data passed to the panic hook is paramaterized, then I'm not sure what the benefit of a `tracing`-vended panic hook is over `std:panic::set_hook`.
- Failing/panicking if `tracing::set_panic_hook` (or another panic hook) is _way_  too opinionated for `tracing`, IMO.

**Pull request merged**